rules/plists_test.bzl

```diff
@@ -0,0 +1,176 @@
+"""
+Tests for the defines.bzl file.
+"""
+
+load("@bazel_skylib//lib:unittest.bzl", "analysistest", "asserts")
+load("//rules:plists.bzl", "process_infoplists")
+
+def _test_process_infoplists_impl(ctx):
+    env = analysistest.begin(ctx)
+    expected_substitutions = ctx.attr.expected_substitutions
+
+    expand_action = analysistest.target_actions(env)[0]
+    actual_substitutions = expand_action.substitutions
+
+    asserts.equals(env, expected_substitutions, actual_substitutions)
+
+    return analysistest.end(env)
+
+process_infoplists_test = analysistest.make(
+    _test_process_infoplists_impl,
+    attrs = {
+        "expected_substitutions": attr.string_dict(),
+    },
+)
+
+def process_infoplists_test_suite(name):
+    """
+    Test suite for process_infoplists.
+
+    Args:
+        name: Name of the test suite.
+    Returns:
+        Labels of the created tests.
+    """
+
+    # Given default xcconfigs
+    xcconfig = {
+        "CUSTOM_PRODUCT_VERSION": "9.9.9",
+    }
+
+    # Given xcconfigs by build setting
+    xcconfig_by_build_setting = {
+        "//:debug": {
+            "CUSTOM_PRODUCT_VERSION": "7.7.7",
+        },
+        "//:release": {
```

Comment:
People can feed this rule code where `xcconfig_by_build_setting`  have different exclusivity than the selects in `infoplists_by_build_setting`.

The related code only works because it only keys on the xcconfig which isn't great from an API perspective

```
 58     for (build_setting, build_setting_xcconfig) in xcconfig_by_build_setting.items():
 59         name_suffix = build_setting_name(build_setting)
 60
```